tonic/src/transport/server/mod.rs

```diff
@@ -360,20 +364,44 @@ impl Server {
         Router::new(self.clone(), svc)
     }
 
-    pub(crate) async fn serve_with_shutdown<S, I, F, IO, IE>(
+    /// TODO(david): docs
```

Comment:
Reminder to self to write docs for this.

Reply:
I assume you plan to include this in this PR?

Reply:
Yes totally!